File: src/main/java/nl/dannyjelsma/soundboardbot/commands/music/QueueCommand.java
package nl.dannyjelsma.soundboardbot.commands.music;

import com.jagrosh.jdautilities.command.Command;
import com.jagrosh.jdautilities.command.CommandEvent;
import com.sedmelluq.discord.lavaplayer.track.AudioTrack;
import nl.dannyjelsma.soundboardbot.audio.SoundManager;
import nl.dannyjelsma.soundboardbot.audio.SoundScheduler;

import java.util.List;

public class QueueCommand extends Command {

    public QueueCommand() {
        name = "queue";
        help = "=queue";
    }

    @Override
    protected void execute(CommandEvent event) {

        SoundScheduler scheduler = SoundManager.getInstance().getScheduler();
        List<AudioTrack> queue = scheduler.getMusicQueue();
        StringBuilder queueStr = new StringBuilder();
        int queueIndex = 0;
        boolean isFirstMsg = true;

        for (AudioTrack track : queue) {
            int currentLength = queueStr.length();
            String line = queueIndex + ". " + track.getInfo().title + " by " + track.getInfo().author + "\n";

            if (currentLength + line.length() >= 1500) {
                String queueList = queueStr.toString();

                if (isFirstMsg) {
                    queueList = "**There are " + queue.size() + " songs in the queue:**\n" + queueList;
                    isFirstMsg = false;
                }

                event.getChannel().sendMessage(queueList).queue();
                queueStr = new StringBuilder();
            } else {
                queueStr.append(line);
            }

            queueIndex += 1;
        }

        if (queueStr.length() > 0) {
            String songList = queueStr.toString();
            event.getChannel().sendMessage(songList).queue();
        }
    }
}

File: src/main/java/nl/dannyjelsma/soundboardbot/commands/music/QueueCommand.kt
package nl.dannyjelsma.soundboardbot.commands.music

import com.jagrosh.jdautilities.command.Command
import com.jagrosh.jdautilities.command.CommandEvent
import nl.dannyjelsma.soundboardbot.audio.SoundManager

class QueueCommand : Command() {
    init {
        name = "queue"
        help = "=queue"
    }

    override fun execute(event: CommandEvent) {
        val queue = SoundManager.scheduler.musicQueue
        var queueStr = StringBuilder()
        var queueIndex = 0
        var isFirstMsg = true

        for (track in queue) {
            val currentLength = queueStr.length
            val line = "$queueIndex. ${track.info.title} by ${track.info.author}"

            if (currentLength + line.length >= 1500) {
                var queueList = queueStr.toString()
                if (isFirstMsg) {
                    queueList = "**There are ${queue.size} songs in the queue:**$queueList"
                    isFirstMsg = false
                }
                event.channel.sendMessage(queueList).queue()
                queueStr = StringBuilder()
            } else {
                queueStr.append(line)
            }

            queueIndex += 1
        }

        if (queueStr.isNotEmpty()) {
            val songList = queueStr.toString()
            event.channel.sendMessage(songList).queue()
        }
    }
}